asserts/database.go

```diff
@@ -264,6 +264,25 @@ func OpenDatabase(cfg *DatabaseConfig) (*Database, error) {
 	}, nil
 }
 
+// WithStackedBackstore returns a new database that adds to the given backstore
+// only but finds in backstore and the base database backstores and
+// cross-checks against all of them.
+// This is useful to cross-check a set of assertions without adding
+// them to the database.
+func (db *Database) WithStackedBackstore(backstore Backstore) *Database {
+	backstores := []Backstore{db.trusted, db.predefined}
+	backstores = append(backstores, backstore)
+	backstores = append(backstores, db.backstores[2:]...)
```

Comment:
This is a little fragile, perhaps add a
```suggestion
        // Ignore the two built-in backstores to avoid re-checking the same thing over.
	backstores = append(backstores, db.backstores[2:]...)
```

Alternatively, since `db.trusted` and `db.predefined` are also stored as distinct fields, you could avoid storing them in backstores altogether and adjust iterating code elsewhere.